pkg/controller/operators/catalog/operator.go

```diff
@@ -1240,8 +1246,17 @@ func transitionInstallPlanState(log *logrus.Logger, transitioner installPlanTran
 			out.Status.Phase = v1alpha1.InstallPlanPhaseFailed
 			return out, err
 		}
-		out.Status.SetCondition(v1alpha1.ConditionMet(v1alpha1.InstallPlanInstalled, &now))
-		out.Status.Phase = v1alpha1.InstallPlanPhaseComplete
+		// Loop over one final time to check and see if everything is good.
+		if !out.Status.NeedsRequeue() {
+			log.Info("Status does not need a requeue!")
+			out.Status.SetCondition(v1alpha1.ConditionMet(v1alpha1.InstallPlanInstalled, &now))
+			out.Status.Phase = v1alpha1.InstallPlanPhaseComplete
+		} else {
+			out.Status.Phase = v1alpha1.InstallPlanPhaseRequeue
+		}
+		return out, nil
+	case v1alpha1.InstallPlanPhaseRequeue:
```

Comment:
If a CRD takes multiple loops to register, won't this cycle Installing -> Requeue -> Installing?

I think we can make it just using `Installing` and remove the `Requeue` phase if we also remove these lines:

```go
	// no changes in status, don't update
	if outInstallPlan.Status.Phase == plan.Status.Phase {
		return
	}
```